src/software/ai/navigator/path_planner/path_planner.h

```diff
@@ -15,8 +17,22 @@
 class PathPlanner
 {
    public:
-    virtual std::optional<std::vector<Point>> findPath(const Point &start,
-                                                       const Point &dest) = 0;
+    /**
+     * Returns a "good" path between start and dest.
+     *
+     * @param start start point
+     * @param dest destination point
+     * @param field field
+     * @param obstacles obstacles to avoid
+     *
+     * @return the optimal path between start and dest
+     * 		if no valid path then return std::monostate
+     * 		if spline navigator then return a Curve
+     * 		if point navigator then return a std::vector of points
+     */
+    virtual std::variant<std::monostate, Curve, std::vector<Point>> findPath(
```

Comment:
Might be convenient to have a typedef like `using PathType = std::variant<Curve, std::vector<Point>>;`

Reply:
ok